nextjs/src/app/reports/[id]/(components)/MapLayers/PoliticalChoropleths.tsx

```diff
@@ -79,16 +77,22 @@ const PoliticalChoropleths: React.FC<PoliticalChoroplethsProps> = ({
         promoteId={tileset.promoteId}
       >
         {/* Fill of the boundary */}
-        <Layer
-          beforeId="road-simple"
-          id={`${tileset.mapboxSourceId}-fill`}
-          source={tileset.mapboxSourceId}
-          source-layer={tileset.sourceLayerId}
-          type="fill"
-          filter={getChoroplethFillFilter(dataByBoundary, tileset)}
-          paint={getChoroplethFill(dataByBoundary, visibility === 'visible')}
-          //layout={{ visibility: delayedVisibility }}
-        />
+        {showChoropleth && (
+          <>
+            <Layer
+              beforeId="road-simple"
+              id={`${tileset.mapboxSourceId}-fill`}
+              source={tileset.mapboxSourceId}
+              source-layer={tileset.sourceLayerId}
+              type="fill"
+              filter={getChoroplethFillFilter(dataByBoundary, tileset)}
+              paint={getChoroplethFill(
+                dataByBoundary,
+                visibility === 'visible'
+              )}
+            />
+          </>
+        )}
```

Comment:
This will change the layer order in ways that might break the visualisation.

There is already a `const visibility` statement further up. Let's just add an extra condition into that instead.

Reply:
done now I think